qa/tasks/ceph_manager.py

```diff
@@ -945,29 +946,37 @@ def test_pool_min_size(self):
                          format(rand=rand_val))
                 if len(self.live_osds) > minup+1 and rand_val < 0.5:
                     # chose to knock out as many OSDs as we can w/out downing PGs
-                    
-                    most_killable = min(len(self.live_osds) - minup, m)
+
+                    most_killable = min(len(self.live_osds) - (minup + 1), m)
                     self.log("chose to kill {n} OSDs".format(n=most_killable))
-                    for i in range(1, most_killable):
+                    for i in range(0, most_killable):
+                        if self.ceph_manager.acting_set_missing(m):
+                            self.log('Some PGs are already missing m shards, we are not killing anymore OSDs!')
+                            break
                         self.kill_osd(mark_out=True)
-                    time.sleep(10)
+                        time.sleep(5) # 5 seconds wait incase PGs want to adjust acting-set.
                     # try a few times since there might be a concurrent pool
                     # creation or deletion
+                    self.log("check for active or peered")
+                    self.log("dead_osds={d}, live_osds={ld}".format(d=self.dead_osds, ld=self.live_osds))
                     with safe_while(
                             sleep=25, tries=5,
                             action='check for active or peered') as proceed:
                         while proceed():
                             if self.ceph_manager.all_active_or_peered():
                                 break
+                            elif self.ceph_manager.acting_set_below_min_size(min_size):
```

Comment:
I don't think I understand what's going on here or with the acting_set_mising checks.  If pgs are below min_size or missing too many shards when they shouldn't be, that should simply be a test failure.

Reply:
@athanatos 
the acting_set_missing checks is needed because we cannot rely on the number of down OSDs alone to determine whether the next one we kill will make the PG go inactive. For example if we kill 3/8 OSDs, there will still be k+1 OSDs left which in theory we should still be able to recover.  However, what if the 3 OSDs belong to an acting set of a PG. So that means 3 shards are missing and the PG would go inactive (If my understanding is correct).

Reply:
Ah, because there may be no PG containing all N of the OSDs you have marked down so far.  It seems like it would a lot simpler to select a PG at random from the pool and then kill the correct number of acting set members on that PG.